client/pkg/transport/listener_test.go

```diff
@@ -573,3 +578,140 @@ func TestSocktOptsEmpty(t *testing.T) {
 		}
 	}
 }
+
+// TestNewListenerWithACRLFile tests when a revocation list is present.
+func TestNewListenerWithACRLFile(t *testing.T) {
+	clientTLSInfo, err := createSelfCertEx(t, "127.0.0.1", x509.ExtKeyUsageClientAuth)
+	if err != nil {
+		t.Fatalf("unable to create client cert: %v", err)
+	}
+	loaded, err := os.ReadFile(clientTLSInfo.CertFile)
+	if err != nil {
+		t.Fatalf("unable to read client cert: %v", err)
+	}
+	block, _ := pem.Decode(loaded)
+	clientCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("unable to parse client cert: %v", err)
+	}
+
+	tests := map[string]struct {
+		expectHandshakeError   bool
+		revokedCertificates    []pkix.RevokedCertificate
+		revocationListContents []byte
+	}{
+		"empty revocation list": {
+			expectHandshakeError: false,
+		},
+		"client cert is revoked": {
+			expectHandshakeError: true,
+			revokedCertificates: []pkix.RevokedCertificate{
+				{
+					SerialNumber:   clientCert.SerialNumber,
+					RevocationTime: time.Now(),
+				},
+			},
+		},
+		"invalid CRL file content": {
+			expectHandshakeError:   true,
+			revocationListContents: []byte("@invalidcontent"),
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			tmpdir := t.TempDir()
+			tlsInfo, err := createSelfCert(t)
+			if err != nil {
+				t.Fatalf("unable to create server cert: %v", err)
+			}
+			tlsInfo.TrustedCAFile = clientTLSInfo.CertFile
+			tlsInfo.CRLFile = filepath.Join(tmpdir, "revoked.r0")
+
+			loaded, err := os.ReadFile(tlsInfo.CertFile)
+			if err != nil {
+				t.Fatalf("unable to read server cert: %v", err)
+			}
+			block, _ := pem.Decode(loaded)
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				t.Fatalf("unable to decode server cert: %v", err)
+			}
+
+			loaded, err = os.ReadFile(tlsInfo.KeyFile)
+			if err != nil {
+				t.Fatalf("unable to read server key: %v", err)
+			}
+			block, _ = pem.Decode(loaded)
+			key, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				t.Fatalf("unable to parse server key: %v", err)
+			}
```

Comment:
It's almost identical to line 631 ~ 639, could you get it included in a common [anonymous/inlined] function?

Reply:
Done. It was used in another instance before, so I pushed the definition of the anonymous function to the top of the test function. Should I consider then defining it as a function (not anonymous)?

Reply:
Looks much better now, thx